openbb_energy/eia/crude_oil/crude_oil.py

```diff
@@ -0,0 +1,212 @@
+"""EIA Crude Oil Import Models."""
+
+import warnings
+from datetime import date as dateType
+from typing import Any, Dict, List, Literal, Optional
+
+from openbb_core.provider.abstract.data import Data
+from openbb_core.provider.abstract.fetcher import Fetcher
+from openbb_core.provider.abstract.query_params import QueryParams
+from pydantic import Field, field_validator
+
+_warn = warnings.warn
+
+CRUDE_OIL_FACET_LIST = ["originId", "originType", "destinationId", "destinationType", "gradeId"]
+
+
+class CrudeOilImportsQueryParams(QueryParams):
+    """Crude Oil Imports query parameters."""
+
+    frequency: Literal["monthly", "annual"] = Field(
+        description="Frequency of the data to be returned.",
+        default="monthly",
+    )
+
+    start_date: Optional[str] = Field(
+        description="Start date of the data to be returned."
+        + " Format: YYYY for annual data, YYYY-MM for monthly data.",
+        default=None,
+        alias="start",
+    )
+
+    end_date: Optional[str] = Field(
+        description="End date of the data to be returned."
+        + " Format: YYYY for annual data, YYYY-MM for monthly data.",
+        default=None,
+        alias="end",
+    )
+
+    filter_by_origin_id: Optional[Literal[
+        "CTY_CA", "CTY_BL", "CTY_NI", "CTY_SY", "CTY_TH", "CTY_KZ", "REG_ME", "CTY_PM", "CTY_CO", "CTY_EK",
+        "CTY_AG", "CTY_AR", "CTY_VM", "CTY_CG", "CTY_GH", "CTY_IR", "CTY_GT", "CTY_IZ", "CTY_AS", "CTY_VE",
+        "CTY_MU", "CTY_IV", "CTY_AL", "CTY_DA", "CTY_TS", "CTY_BB", "CTY_CF", "CTY_EG", "CTY_KU", "CTY_YM",
+        "REG_EU", "OPN_Y", "CTY_AE", "CTY_LY", "CTY_NO", "CTY_MR", "CTY_NL", "REG_OA", "REG_AF", "OPN_N",
+        "CTY_SP", "CTY_OD", "CTY_PP", "CTY_CH", "CTY_GB", "CTY_AJ", "CTY_AO", "CTY_BH", "CTY_BR", "CTY_MX",
+        "CTY_RS", "CTY_SA", "CTY_ID", "REG_CA", "CTY_CD", "CTY_EC", "CTY_CM", "CTY_TD", "CTY_UK", "CTY_BX",
+        "REG_AP", "CTY_GY", "CTY_MY", "CTY_PE", "CTY_QA", "CTY_IT", "CTY_SF", "REG_EA", "WORLD", "CTY_BF", "CTY_SG"
+    ]] = Field(
+        description="Filter by origin ID. You can provide a comma separated list of origin IDs."
+        + " Examples: CTY_CA (Canada), CTY_SA (Saudi Arabia), REG_ME (Middle East).",
+        default=None,
+        alias="originId",
+    )
+
+    filter_by_origin_type: Optional[Literal["REG", "CTY", "OPN", "WORLD"]] = Field(
+        description="Filter by origin type. You can provide a comma separated list of origin types."
+        + " Choose from: REG (Region), CTY (Country), OPN (OPEC/non-OPEC), WORLD (World).",
+        default=None,
+        alias="originType",
+    )
+
+    filter_by_destination_id: Optional[Literal[
+        "RF_394", "RF_528", "RF_530", "RF_2403", "RF_310", "RF_148", "RF_506", "RF_83", "RF_98", "RF_289",
```

Comment:
This list is very long. Come up with a clever way to type-hint this field without the need to have this list here.